src/Native/System.Native/pal_networking.cpp

```diff
@@ -2094,7 +2094,10 @@ static Error WaitForSocketEventsInner(int32_t port, SocketEvent* buffer, int32_t
     assert(numEvents != 0);
     assert(numEvents < *count);
 
-    if (sizeof(epoll_event) < sizeof(SocketEvent))
+    size_t sizeof_epoll_event = sizeof(epoll_event);
+    size_t sizeof_socket_event = sizeof(SocketEvent);
```

Comment:
Hmm, someone is likely to put this back as it was without a comment. Since clang-3.5+ seems forgiving to the extent that we want, maybe we should just test for < 3.5 in CMakeLists.txt and add  -W-no-unreachable-code in that case.